### PR TITLE
Apply a solution filter (revert revert)

### PR DESCRIPTION
**Pull request by Forgind**
This reverts commit 2e4c1867b08f0fe9ae2591cde6cb73ab304efeff.

Original slnf change was #4899

Revert was #5142. #4899 was reverted because it created a System.Text.Json dependency that has now been corrected.

### CODE REVIEW

File: src/Build/Construction/Solution/SolutionFile.cs [hunk 1]
```diff
@@ -462,6 +509,30 @@ internal void ParseSolution()
...
+                foreach (ProjectInSolution project in _projectsInOrder)
+                {
+                    projectPaths.Add(project.RelativePath);
+                }
+                foreach (string project in _solutionFilter)
+                {
+                    if (!projectPaths.Contains(project))
+                    {
+                        ProjectFileErrorUtilities.VerifyThrowInvalidProjectFile
+                        (
```
Comment (danmosemsft) [inline src/Build/Construction/Solution/SolutionFile.cs:524]: is this missing the first parameter (false)?

Comment (danmosemsft) [reply src/Build/Construction/Solution/SolutionFile.cs:524]: oops, no idea how I ended up commenting on an old PR :)

Comment (Forgind) [reply src/Build/Construction/Solution/SolutionFile.cs:524]: This theoretically will still go in if System.Text.Json starts working. This PR also specifies what VerifyThrowInvalidProjectFile should do if it's missing the first parameter: assume false.

Comment (danmosemsft) [reply src/Build/Construction/Solution/SolutionFile.cs:524]: Ah - I thought the overloads that assume false were named `ThrowInvalidProjectFile()`

**Pull request merged**